src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h

```diff
@@ -301,6 +302,13 @@ using addr_type = uint64_t;
       bool readEmuSettingsJsonFile(const std::string& emuSettingsFilePath);
 
     private:
+      std::thread mMessengerThread;
+      std::thread mHostMemAccessThread;
+      std::atomic_bool mMessengerThreadStarted;
+      std::atomic_bool mHostMemAccessThreadStarted;
```

Comment:
```suggestion
      std::atomic<bool> mMessengerThreadStarted;
      std::atomic<bool> mHostMemAccessThreadStarted;
```

Reply:
Though atomic_bool serves the purpose, updating as per comment.

Reply:
We are trying to make software better, from "serves the purpose" or "oh, it works on my machine today" towards software engineering. ;-)